### PR TITLE
Istio 1.6 tested

### PR DESCRIPTION
**Pull request by Sodman**
I've tested this with gloo 1.3.20 and 1.4.0-beta13, everything works as expected.

I'd like some input on how best to layout these docs though, as this page has grown quite a bit. A whole new section is in some ways overkill when we're just bumping the istio version in a few urls, and changing a healthcheck port by 1 digit.

Possible Options:

1. Leave as-is in this PR, have a smaller 1.6 section that has a shortest-path walkthrough and calls back to the 1.5 section if the user wants more details
2. Combine both into an "Istio 1.5.x & 1.6.x" section, favoring 1.6.0 in the URLs and calling out any other differences in warning bubbles.
3. Break out the documentation by Istio version so we have a dedicated page for 1.4.x, 1.5.x, 1.6.x, etc. The pages may or may not vary much between versions (eg 1.4->1.5 is a decent change, 1.5->1.6 is very minor changes)
4. Copy the extra detailed config section from 1.5 to 1.6, changing just the istio version and port number where needed

### CODE REVIEW

**Comment by Sodman** [comment]
/kick Regression test flake - https://console.cloud.google.com/cloud-build/builds/79395cbf-5656-4040-b625-0e83911f8fc1;step=14

**Comment by Sodman** [comment]
/skip-changelog

pushed unsaved local doc change

File: docs/content/guides/integrations/service_mesh/gloo_istio_mtls.md [hunk 1]
```diff
@@ -736,3 +736,86 @@ Test:
...
+Install gloo:
+```
+glooctl install gateway
+```
+or with helm:
+```
+kubectl create ns gloo-system; helm install --namespace gloo-system --version 1.3.20 gloo gloo/gloo
+```
+
+Patch gloo's envoy deployment. Note that in this example we are using kind, so our parameters are different than outlined above (note that you may need to update the version of the gateway-proxy container to match your version. in this example it is set to `gloo-envoy-wrapper:1.3.20`):
```
Comment (kdorosh) [inline docs/content/guides/integrations/service_mesh/gloo_istio_mtls.md:779]: nit, capitalization
```suggestion
Patch gloo's envoy deployment. Note that in this example we are using kind, so our parameters are different than outlined above (note that you may need to update the version of the gateway-proxy container to match your version. In this example it is set to `gloo-envoy-wrapper:1.3.20`):
```

File: docs/content/guides/integrations/service_mesh/gloo_istio_mtls.md [hunk 2]
```diff
@@ -736,3 +736,86 @@ Test:
...
...ue":"cluster.local"},{"name":"INSTANCE_IP","valueFrom":{"fieldRef":{"fieldPath":"status.podIP"}}},{"name":"SERVICE_ACCOUNT","valueFrom":{"fieldRef":{"fieldPath":"spec.serviceAccountName"}}},{"name":"HOST_IP","valueFrom":{"fieldRef":{"fieldPath":"status.hostIP"}}},{"name":"ISTIO_META_POD_NAME","valueFrom":{"fieldRef":{"apiVersion":"v1","fieldPath":"metadata.name"}}},{"name":"ISTIO_META_CONFIG_NAMESPACE","valueFrom":{"fieldRef":{"apiVersion":"v1","fieldPath":"metadata.namespace"}}}],"imagePullPolicy":"IfNotPresent","readinessProbe":{"failureThreshold":30,"httpGet":{"path":"/healthz/ready","port":15021,"scheme":"HTTP"},"initialDelaySeconds":1,"periodSeconds":2,"successThreshold":1,"timeoutSeconds":1},"volumeMounts":[{"mountPath":"/var/run/secrets/istio","name":"istiod-ca-cert"},{"mountPath":"/etc/istio/proxy","name":"istio-envoy"},{"mountPath":"/etc/istio-certs/","name":"istio-certs"}]}],"volumes":[{"configMap":{"defaultMode":420,"name":"gateway-proxy-envoy-config"},"name":"envoy-config"},{"name":"istio-certs","emptyDir":{"medium":"Memory"}},{"name":"istiod-ca-cert","configMap":{"defaultMode":420,"name":"istio-ca-root-cert"}},{"emptyDir":{"medium":"Memory"},"name":"istio-envoy"}]}}}}'
```
Comment (kdorosh) [inline docs/content/guides/integrations/service_mesh/gloo_istio_mtls.md:781]: can we add a collapsible section with the json from this patch that's formatted for readability so users can get an idea of what the patch is doing to the cluster?

Comment (kdorosh) [reply docs/content/guides/integrations/service_mesh/gloo_istio_mtls.md:781]: noticed we do the same thing across our other examples. i think there's room for this improvement across the whole page. but no need for that to block this PR.

Comment (kdorosh) [reply docs/content/guides/integrations/service_mesh/gloo_istio_mtls.md:781]: opened https://github.com/solo-io/gloo/issues/3116 to track that work

**Pull request merged**